rspirv/dr/build/mod.rs

```diff
@@ -749,36 +749,9 @@ impl Builder {
         id
     }
 
-    /// Appends an OpConstant instruction with the given 32-bit float `value`.
+    /// Appends an OpConstant instruction with the given 32-bit bit pattern `value`.
     /// or the module if no block is under construction.
-    pub fn constant_f32(&mut self, result_type: spirv::Word, value: f32) -> spirv::Word {
-        let id = self.id();
-        let inst = dr::Instruction::new(
-            spirv::Op::Constant,
-            Some(result_type),
-            Some(id),
-            vec![dr::Operand::LiteralFloat32(value)],
-        );
-        self.module.types_global_values.push(inst);
-        id
-    }
-
-    /// Appends an OpConstant instruction with the given 64-bit float `value`.
-    pub fn constant_f64(&mut self, result_type: spirv::Word, value: f64) -> spirv::Word {
-        let id = self.id();
-        let inst = dr::Instruction::new(
-            spirv::Op::Constant,
-            Some(result_type),
-            Some(id),
-            vec![dr::Operand::LiteralFloat64(value)],
-        );
-        self.module.types_global_values.push(inst);
-        id
-    }
-
-    /// Appends an OpConstant instruction with the given 32-bit integer `value`.
-    /// or the module if no block is under construction.
-    pub fn constant_u32(&mut self, result_type: spirv::Word, value: u32) -> spirv::Word {
+    pub fn constant_int32(&mut self, result_type: spirv::Word, value: u32) -> spirv::Word {
```

Comment:
Nitpicking: could we also name these `constant_32bit` and `constant_64bit`? Same for the other types, so that it's a bit more clear that it's a generic "array of bytes" rather than a (float interpreted as) integer.

That also matches the updated doc-comment on this function (and we could have a similar doc-comment on the `dr::Operand` enum variant).

Reply:
Decided to go with `bit32` and `bit64` instead to avoid any names starting with numbers.

Reply:
Thanks! I didn't expect it to be such an intrusive change, but worth it IMO. Will now have to wait for a maintainer to say the same :)